File: src/net/owatz/CardDeck.java
package net.owatz;

import java.util.ArrayList;

public class CardDeck {

    public ArrayList<Card> deck;

    public CardDeck() {
        newDeck();
    }

    public void newDeck() {

        deck = new ArrayList<>();
        for (int value=1; value<=13; value++) {
            for (int color=1; color<=4; color++) {
                deck.add(new Card(color, value, (value-1)*4+color-1));
            }
        }
    }

    public Card getRandomCard() {

        int randNum = (int)(Math.random() * (deck.size()));
        Card randCard = deck.get(randNum);
        deck.remove(randNum);
        return randCard;
    }
}

File: src/CardDeck.ts
import {Card} from "./Card";

export class CardDeck {
  deck: Card[];

  constructor() {
    this.deck = [];
    for (let value=1; value<=13; value++) {
      for (let color=0; color<=3; color++) {
        this.deck.push(new Card(color, value))
      }
    }
    CardDeck.shuffleArray(this.deck);
  }

  getRandomCard():Card {
    return this.deck.pop();
  }

  /**
   * https://en.wikipedia.org/wiki/Fisher–Yates_shuffle
   * https://stackoverflow.com/a/12646864
   * @param array to shuffle
   */
  private static shuffleArray<T>(array: Array<T>) {
    for (let i = array.length - 1; i > 0; i--) {
      const j = Math.floor(Math.random() * (i + 1));
      [array[i], array[j]] = [array[j], array[i]];
    }
  }
}